types/relay-compiler/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for relay-compiler 5.0
+// Project: https://relay.dev
+// Definitions by: n1ru4l <https://github.com/n1ru4l>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+import { GraphQLCompilerContext } from './lib/GraphQLCompilerContext';
+import * as ASTConvert from './lib/ASTConvert';
+
+import * as Parser from './lib/RelayParser';
+import * as Printer from './lib/GraphQLIRPrinter';
+import ConsoleReporter from './lib/GraphQLConsoleReporter';
+import MultiReporter = require('./lib/GraphQLMultiReporter');
+
+declare var transformASTSchema: typeof ASTConvert.transformASTSchema;
+
+export { GraphQLCompilerContext, ASTConvert, transformASTSchema, Parser, Printer, ConsoleReporter, MultiReporter };
```

Comment:
Is this missing few more typings?:

```js
// a.js
const x = import("relay-compiler");
console.log(x);
```

results in 
```
> node a.js
ASTConvert:
   { convertASTDocuments: [Function: convertASTDocuments],
     convertASTDocumentsWithBase: [Function: convertASTDocumentsWithBase],
     extendASTSchema: [Function: extendASTSchema],
     transformASTSchema: [Function: transformASTSchema] },
  CodegenDirectory:
   { [Function: CodegenDirectory]
     combineChanges: [Function: combineChanges],
     hasChanges: [Function: hasChanges],
     printChanges: [Function: printChanges],
     getAddedRemovedFiles: [Function: getAddedRemovedFiles],
     sourceControlAddRemove: [Function: sourceControlAddRemove] },
  CodegenRunner: [Function: CodegenRunner],
  CodegenWatcher:
   { queryDirectories: [Function: queryDirectories],
     queryFiles: [Function: queryFiles],
     queryFilepaths: [Function: queryFilepaths],
     watch: [Function: watch],
     watchFiles: [Function: watchFiles],
     watchCompile: [Function: watchCompile] },
  CompilerContext: [Function: GraphQLCompilerContext],
  ConsoleReporter: [Function: GraphQLConsoleReporter],
  DotGraphQLParser:
   { parseFile: [Function: parseFile],
     getParser: [Function: getParser] },
  ASTCache: [Function: ASTCache],
  IRTransformer: { transform: [Function: transform] },
  IRVisitor: { visit: [Function: visitIR] },
  Printer:
   { print: [Function: print],
     printField: [Function: printField],
     printArguments: [Function: printArguments],
     printDirectives: [Function: printDirectives] },
  Profiler:
   { enable: [Function: enable],
     getTraces: [Function: getTraces],
     run: [Function: run],
     asyncContext: [Function: asyncContext],
     waitFor: [Function: waitFor],
     instrument: [Function: instrument],
     instrumentAsyncContext: [Function: instrumentAsyncContext],
     instrumentWait: [Function: instrumentWait],
     start: [Function: start],
     startWait: [Function: startWait],
     end: [Function: end] },
  Rollout: { set: [Function: set], check: [Function: check] },
  SchemaUtils:
   { assertTypeWithFields: [Function: assertTypeWithFields],
     canHaveSelections: [Function: canHaveSelections],
     generateIDField: [Function: generateIDField],
     getNullableType: [Function: getNullableType],
     getRawType: [Function: getRawType],
     getSingularType: [Function: getSingularType],
     hasID: [Function: hasID],
     implementsInterface: [Function: implementsInterface],
     isAbstractType: [Function: isAbstractType],
     isClientDefinedField: [Function: isClientDefinedField],
     isExecutableDefinitionAST: [Function: isExecutableDefinitionAST],
     isSchemaDefinitionAST: [Function: isSchemaDefinitionAST],
     isServerDefinedField: [Function: isServerDefinedField],
     isUnionType: [Function: isUnionType],
     mayImplement: [Function: mayImplement] },
  SourceControlMercurial: { addRemove: [Function: addRemove] },
  WatchmanClient:
   { [Function: GraphQLWatchmanClient] isAvailable: [Function: isAvailable] },
  filterContextForNode: [Function: filterContextForNode],
  getIdentifierForArgumentValue: [Function: getIdentifierForArgumentValue],
  getNormalizationOperationName: [Function: getNormalizationOperationName],
  getLiteralArgumentValues: [Function: getLiteralArgumentValues],
  isEquivalentType: [Function: isEquivalentType],
  nullthrows: [Function: nullthrows],
  Parser:
   { parse: [Function: parse], transform: [Function: transform] },
  Validator:
   { GLOBAL_RULES:
      [ [Function: KnownArgumentNames],
        [Function: NoUnusedVariables],
        [Function: UniqueArgumentNames],
        [Function: UniqueFragmentNames],
        [Function: UniqueInputFieldNames],
        [Function: UniqueOperationNames],
        [Function: UniqueVariableNames] ],
     LOCAL_RULES:
      [ [Function: FragmentsOnCompositeTypes],
        [Function: KnownTypeNames],
        [Function: LoneAnonymousOperation],
        [Function: PossibleFragmentSpreads],
        [Function: ValuesOfCorrectType],
        [Function: VariablesAreInputTypes],
        [Function: DisallowIdAsAliasValidationRule] ],
     validate: [Function: validateOrThrow] },
  CodeGenerator: { generate: [Function: generate] },
  FlowGenerator:
   { generate: [Function: generate],
     transforms:
      [ [Function: relayRelayDirectiveTransform],
        [Function: relayMaskTransform],
        [Function: relayMatchTransform],
        [Function: flattenTransform],
        [Function: relayRefetchableFragmentTransform] ] },
  GraphQLCompilerContext: [Function: GraphQLCompilerContext],
  FileWriter: { writeAll: [Function: writeAll] },
  IRTransforms:
   { commonTransforms:
      [ [Function: relayConnectionTransform],
        [Function: relayRelayDirectiveTransform],
        [Function: relayMaskTransform],
        [Function: relayMatchTransform],
        [Function: relayRefetchableFragmentTransform] ],
     codegenTransforms:
      [ [Function: skipUnreachableNodeTransform],
        [Function: relaySplitMatchTransform],
        [Function: inlineFragmentsTransform],
        [Function: clientExtensionTransform],
        [Function: flattenTransform],
        [Function: skipRedundantNodesTransform],
        [Function: relayGenerateTypeNameTransform],
        [Function: filterDirectivesTransform],
        [Function: refineOperationVariablesTransform] ],
     fragmentTransforms:
      [ [Function: clientExtensionTransform],
        [Function: relayFieldHandleTransform],
        [Function: flattenTransform],
        [Function: skipRedundantNodesTransform] ],
     printTransforms:
      [ [Function: clientExtensionTransform],
        [Function: skipClientExtensionTransform],
        [Function: skipUnreachableNodeTransform],
        [Function: flattenTransform],
        [Function: relayGenerateTypeNameTransform],
        [Function: relaySkipHandleFieldTransform],
        [Function: filterDirectivesTransform],
        [Function: refineOperationVariablesTransform] ],
     queryTransforms:
      [ [Function: relayApplyFragmentArgumentTransform],
        [Function: relayGenerateIDFieldTransform],
        [Function: relayDeferStreamTransform],
        [Function: testOperationDirective] ],
     schemaExtensions:
      [ '\n  directive @connection(\n    key: String!\n    filters: [String]\n    handler: String\n    dynamicKey_UNSTABLE: String\n  ) on FIELD\n\n  directive @stream_connection(\n    key: String!\n    filters: [String]\n    handler: String\n    label: String!\n    initial_count: Int!\n    if: Boolean = true\n    dynamicKey_UNSTABLE: String\n  ) on FIELD\n',
        '\n  directive @match on FIELD\n\n  directive @module(\n    name: String!\n  ) on FRAGMENT_SPREAD\n',
        '\ndirective @relay(\n  # Marks a connection field as containing nodes without \'id\' fields.\n  # This is used to silence the warning when diffing connections.\n  isConnectionWithoutNodeID: Boolean,\n\n  # Marks a fragment as intended for pattern matching (as opposed to fetching).\n  # Used in Classic only.\n  pattern: Boolean,\n\n  # Marks a fragment as being backed by a GraphQLList.\n  plural: Boolean,\n\n  # Marks a fragment spread which should be unmasked if provided false\n  mask: Boolean = true,\n\n  # Selectively pass variables down into a fragment. Only used in Classic.\n  variables: [String!],\n) on FRAGMENT_DEFINITION | FRAGMENT_SPREAD | INLINE_FRAGMENT | FIELD\n',
        '\n  directive @refetchable(\n    queryName: String!\n  ) on FRAGMENT_DEFINITION\n',
        'directive @relay_test_operation on QUERY | MUTATION | SUBSCRIPTION' ] },
  JSModuleParser:
   { getParser: [Function: getParser],
     getFileFilter: [Function: getFileFilter],
     parseFile: [Function: parseFile],
     parseFileWithSources: [Function: parseFileWithSources] },
  MultiReporter: [Function: GraphQLMultiReporter],
  Runner: [Function: CodegenRunner],
  compileRelayArtifacts: [Function: compileRelayArtifacts],
  formatGeneratedModule: [Function: formatGeneratedModule],
  convertASTDocuments: [Function: convertASTDocuments],
  transformASTSchema: [Function: transformASTSchema],
  getReaderSourceDefinitionName: [Function: getReaderSourceDefinitionName],
  getSourceDefinitionName: [Function: getSourceDefinitionName],
  writeRelayGeneratedFile: [Function: writeRelayGeneratedFile] }
```

Reply:
Is it necessary to add all types for getting this merged? Currently, I only created types for the parts that I need.

Reply:
No but i wanted to point out to ensure it was intentional